### ISSUE TITLE
Queue not restored on start?

### ISSUE THREAD

**Issue by Gwend4l** [gnumdk/lollypop#534]
Using version 0.9.91. I have enabled "restore state at startup" in the preferences.

When I queue multiple albums, if I quit and restart lollypop, it looks like that the only remaining album remaining in the queue is the one from the track that I was playing... All other albums have disappeared from the queue after the restart.

**Comment by d9h02f** [comment]
#533

**Comment by gnumdk** [comment]
It restore current playlist but not perfect I know.
I will not restore queue.

**Comment by Gwend4l** [comment]
Ah, I thought it was a bug. Sorry about the report!

Indeed the current playlist is restored, that's really nice. However I miss two things with playlists:

- If we are playing a playlist, the queue UI is indeed "replaced" by the content of the playlist. However, if we add an album to the playlist while we are playing the playlist, this album does not appear in the queue UI. In other words, it seems that there is no synchronization between the (modifiable) current playlist content and the "playlist queue UI" content?
- I really really love the "+" button that can be used when hovering on an album. Yet it adds the album to the queue, and not to a playlist. Could it be envisioned to be able to use the "+" to add the album at the end of the current playlist instead?

In other words: I would actually prefer to never rely on the queue, and only rely on playlists :D

**Comment by gnumdk** [comment]
>it seems that there is no synchronization between the (modifiable) current playlist content and the >playlist queue UI" content?

Yes, I'm currently trying to fix this one.

>Could it be envisioned to be able to use the "+" to add the album at the end of the current playlist >instead?
I tried but was really ressource expensive due to internal design...

**Comment by Gwend4l** [comment]
Ok I understand :). Well if someday you manage to have a "default" playlist, I'll be even happier with lollypop :). Good luck!

Also, I really like the album-oriented design of lollypop. If there was a way to create albums-only playlists similarly to the albums in the queue (along with an easy way to change the order of the albums in the playlist without having to bother about songs), I would be thrilled.

Thanks for all your work!

**Comment by gnumdk** [comment]
@Gwend4l Your last already exists in Lollypop. Just add albums by clicking the [+]/[-] overlay icon. And manage them at the same place than you go for queue...

**Comment by Gwend4l** [comment]
You are talking about the queue, right? I was suggesting to have the same feature (ie album drag and drop) but when editing playlists. But maybe I misunderstood?

**Comment by gnumdk** [comment]
No, I'm not talking about the queue.

**Comment by gnumdk** [comment]
The list menu button in toolbar works like this:
- If something in the queue, show the queue management widget
- If playing a playlist, show the playlists management widget
- If playlist albums, show the albums management widget
